llvm/lib/Target/X86/X86InstrInfo.cpp

```diff
@@ -8832,6 +8832,11 @@ bool X86InstrInfo::isSchedulingBoundary(const MachineInstr &MI,
       Opcode == X86::PLDTILECFGV)
     return true;
 
+  // Frame setup and destory can't be scheduled around.
+  if (MI.getFlag(MachineInstr::FrameSetup) ||
+      MI.getFlag(MachineInstr::FrameDestroy))
```

Comment:
Should we put it in TargetInstrInfo::isSchedulingBoundary?

Reply:
Lots of arm tests would fail.

Reply:
If arm accepts scheduling with frame setup and destory, why can't for X86?

Reply:
Failed arm tests: https://buildkite.com/llvm-project/github-pull-requests/builds/75530#01904eb7-13d0-451e-b1da-bcb86cd44b60

```
.seh_savexmm %xmm15, 544
.seh_endprologue
```
For X86, most of frame-setup instruction is related to SEH exception restore.
Changing part of its order does not matter. e.g. two independent register save instructions.
Part of them does matter. e.g. .seh_pushreg, moving .seh_endprologue which may affect prologue size. 

Anyway, rescheduling frame-setup instruction does not look like profitable for X86.

Reply:
I can't see the reason from the log, but I'm good with this change as long as no performance regression for X86.